### PR TITLE
[DX-1682] Release Notes 5.5.1 Gateway

### PR DESCRIPTION
**Pull request by dcs3spp**
### For internal users - Please add a Jira DX PR ticket to the subject!
[DX-1682](https://tyktech.atlassian.net/browse/DX-1682)

---

### Preview Link

[preview](https://deploy-preview-5398--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/release-notes/version-5.5)

### Description

Add Gateway release notes for Gateway 5.5.1

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labeled** the PR

[DX-1682]: https://tyktech.atlassian.net/browse/DX-1682?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dcs3spp** [comment]
@caroltyk Please check Tyk charts version associated with release notes 5.5.1 is correct and update accordingly.

After this PR has been published please create a separate PR to include a download link for Tyk Charts when released.

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Added change log content, moved Release Highlights to the top of each section for improved visibility. Just needs to have release date added and downloads section to be completed.

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.5.md [hunk 1]
```diff
@@ -2,29 +2,138 @@
...
+We have introduced two new options in the `http_server_options` [Gateway configuration]({{< ref "tyk-oss-gateway/configuration#http_server_options" >}}) that will enforce prefix and/or suffix matching when Tyk performs checks on whether middleware or other logic should be applied to a request:
+
+- `enable_path_prefix_matching` ensures that the start of the request path must match the path defined in the API definition
+- `enable_path_suffix_matching` ensures that the end of the request path must match the path defined in the API definition
+- combining `enable_path_prefix_matching` and `enable_path_suffix_matching` will ensure an exact (explicit) match is performed
+
+These configuration options provide control to avoid unintended matching of paths from Tyk's default *wildcard* match. Use of regex special characters when declaring the endpoint path in the API definition will automatically override these settings for that endpoint.
+
+Tyk recommends that exact matching is employed, but both options default to `false` to avoid introducing a breaking change for existing users.
+</details>
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.5.md]:
```suggestion

The example Gateway configuration file `tyk.conf.example` has been updated to set the recommended *exact matching* with:

 - `http_server_options.enable_path_prefix_matching = true`
 - `http_server_options.enable_path_suffix_matching = true`
 - `http_server_options.enable_strict_routes = true`
 </details>
```

**Review by andyo-tyk — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.5.md [hunk 2]
```diff
@@ -2,29 +2,144 @@
...
 ## Support Lifetime
-<!-- Required. replace X.Y with this release and set the correct quarter of the year -->
+
 Our minor releases are supported until our next minor comes out.
 
 ---
 
+## 5.5.1 Release Notes
+
+### Release Date ?? September 2024
```
Comment (letzya) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.5.md]:
```suggestion
### Release Date 18 September 2024
```

**Review by carolwkc — APPROVED** [review]
LGTM

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.5.md [hunk 3]
```diff
@@ -2,29 +2,144 @@
...
 ## Support Lifetime
-<!-- Required. replace X.Y with this release and set the correct quarter of the year -->
+
 Our minor releases are supported until our next minor comes out.
 
 ---
 
+## 5.5.1 Release Notes
+
+### Release Date 24 September 2024
```
Comment (letzya) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.5.md]:
```suggestion
### Release Date 26 September 2024
```

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**